### PR TITLE
Updates to dependencies and GitHub actions, and some code clean-up

### PR DESCRIPTION
**Pull request by surilindur**
Here is a somewhat random assortment of various changes, apologies for the mess. I can split the pull request up if needed.

The following changes were made to the actual library itself, in addition to bumping the dependency versions:
* Migrate from TSLint to ESLint and use `@rubensworks/eslint-config` for the rules.
* Migrate from `minimist` to `yargs` for argument parsing, and rewrite the CLI script accordingly.
* Migrate from `pre-commit` to `husky` and remove the dependency on NPM CLI from the scripts.
* Remove `abort-controller` polyfill, because it seems to be supported by everything since quite a few versions now according to [the MDN chart](https://developer.mozilla.org/en-US/docs/Web/API/AbortController#browser_compatibility).
* Add unit test for `fetchRawStream` to ensure the AbortController signal actually works. This should help make sure the AbortController itself is also present in all the tested Node versions.

I have tested these changes with some manual queries on a Comunica SPARQL endpoint with the CLI, and by using the version from this branch with the sparql-benchmark-runner.js library, and it seems to work.

Any feedback would be welcome! I will keep this as a draft until I have addressed most of the issues that will likely be pointed out. I also need to check if anything is needed here to fix the undici body timeout or connection closed errors, probably not.

### CODE REVIEW

**Comment by surilindur** [comment]
I split the GitHub Actions changes into another PR at #75 and removed some unnecessary changes from this here. This should be ready for review now, but I am not sure if the CI will pass before the other PR is merged.

File: package.json [hunk 1]
```diff
@@ -33,57 +32,49 @@
...
+    "sparqljson-parse": "^2.0.0",
+    "sparqlxml-parse": "^2.0.0",
+    "stream-to-string": "^1.0.0",
+    "yargs": "^17.0.0"
   },
   "devDependencies": {
-    "@types/jest": "^29.5.4",
-    "@types/minimist": "^1.2.0",
-    "@types/n3": "^1.10.3",
+    "@rubensworks/eslint-config": "^2.0.0",
```
Comment (rubensworks) [inline package.json:57]: I released v3 recently, perhaps we can update to that one already (see Comunica next/major if you want an example)?

**Comment by surilindur** [comment]
This is now ready for review! I had some issues with the CLI test using Jena Docker image as a SPARQL endpoint, but I finally managed to fix it by adding basic auth support to the CLI, assigning the admin password as an environment variable, and setting up the container as a service according to the [GitHub documentation](https://docs.github.com/en/actions/using-containerized-services/about-service-containers).

**Comment by rubensworks** [comment]
Thanks for the PR!

It's a bit unfortunate that we can't easily test the docker stuff locally anymore using the scripts, but I can live with it.

Since the basic auth + docker stuff and the general cleanup stuff are two distinct things, could you either rewrite the commit history so these end up as 2 separate commits, or split them up into 2 PRs?

**Comment by surilindur** [comment]
The original reason the CLI test with Docker was failing seemed to have been the missing `--query` to declare that the argument is actually the query. After fixing that, it works with the bash scripts just fine.

Since the current changes here do not seem to require a major version bump, and with the CLI test working, I will skip the basic auth changes since no one will probably use it anyway and the CI changes that I made with them are not actually needed beyond clean-up reason. :slightly_smiling_face:

**Comment by rubensworks** [comment]
> The original reason the CLI test with Docker was failing seemed to have been the missing --query to declare that the argument is actually the query. After fixing that, it works with the bash scripts just fine.

Isn't that a problem with this PR then? Because the `-q` option used to be optional before this PR.

> I will skip the basic auth changes since no one will probably use it anyway

Since you did the effort, I don't see a problem with adding support for it :-)

**Comment by surilindur** [comment]
> Isn't that a problem with this PR then? Because the `-q` option used to be optional before this PR.

Actually, that is a good point. :grimacing: I will make it optional again. Thank you. I will also look into making a PR for the basic auth, then.

**Comment by rubensworks** [comment]
Thanks!

**Pull request merged**